conan/tools/gnu/makedeps.py

```diff
@@ -0,0 +1,216 @@
+                "FRAMEWORKS": None,
+                "REQUIRES": None,
+            }
+
+            for var, prefix_var in common_variables.items():
+                cppinfo_value = getattr(cpp_info, var.lower())
+                # Use component cpp_info info when does not provide any value
+                if not cppinfo_value and hasattr(cpp_info, "components"):
+                    cppinfo_value = [self._var_ref(f"CONAN_{var}_{_makeify(var_id)}_{_makeify(name)}") for name, obj in cpp_info.components.items() if getattr(obj, var.lower())]
+                    # avoid repeating same prefix twice
+                    prefix_var = None
+                if "FLAGS" in var:
+                    cppinfo_value = [var.replace('"', '\\"') for var in cppinfo_value]
+                _var(var, cppinfo_value, prefix_var)
+
+        # Top-of-file banner
+        line_buffer.append(self._title)
+        preamble = _divert()
+
+        body = []
+        dep_name_list = []
+
+        host_req = self._conanfile.dependencies.host
+        build_req = self._conanfile.dependencies.build  # tool_requires
+        test_req = self._conanfile.dependencies.test
+
+        for require, dep in list(host_req.items()) + list(build_req.items()) + list(test_req.items()):
+            # Require is not used at the moment, but its information could be used,
+            # and will be used in Conan 2.0
+            # Filter the build_requires not activated with PkgConfigDeps.build_context_activated
+            if require.build:
+                continue
+
+            dep_id = _makeify(require.ref.name)
+            dep_name_list.append(require.ref.name)
+
+            root = dep.recipe_folder if dep.package_folder is None else dep.package_folder
+
+            line_buffer.extend([
+                "",
+                f"# {require.ref.name}/{require.ref.version}",
+                f"# (" + ("direct" if require.direct else "indirect") + " dependency)",
+                ""
+            ])
+
+            _output_var(f"CONAN_NAME_{dep_id}", require.ref.name)
+            _output_var(f"CONAN_VERSION_{dep_id}", require.ref.version)
+            _output_var(f"CONAN_ROOT_{dep_id}", root)
+
+            _output_cpp_info_vars(dep_id, dep.cpp_info, root, dep_id, True, dep)
+
+            dep_section = _divert()
+            comp_name_list = []
+
+            for comp_name, component in dep.cpp_info.get_sorted_components().items():
+                line_buffer.append(f"# {require.ref.name}::{comp_name}")
+                line_buffer.append("")
+                comp_id = _makeify(comp_name)
+                comp_name_list.append(comp_name)
+                _output_cpp_info_vars(f"{dep_id}_{comp_id}", component, root, dep_id, False, dep)
+
+            comp_section = _divert()
+
+            line_buffer.extend(dep_section)
+            _output_var(f"CONAN_COMPONENTS_{dep_id}", comp_name_list)
+            line_buffer.extend(comp_section)
+
+            body.extend(_divert())
```

Comment:
This rendering approach reads a bit complicated. I'd prefer if there would be some visibility over the final result template.